extended-tracer/src/main/java/io/opentelemetry/contrib/tracer/Tracing.java

```diff
@@ -0,0 +1,269 @@
+    return ExtendedTracer.create(GlobalOpenTelemetry.getTracer("service"));
+  }
+
+  /**
+   * Creates a new span builder with the given span name.
+   *
+   * @param spanName the span name
+   * @return the span builder
+   */
+  public static ExtendedSpanBuilder withSpan(String spanName) {
+    return Tracing.serviceTracer().spanBuilder(spanName);
+  }
+
+  /**
+   * Marks a span as error.
+   *
+   * @param span the span
+   * @param description what went wrong
+   */
+  public static void setSpanError(Span span, String description) {
+    span.setStatus(StatusCode.ERROR, description);
+  }
+
+  /**
+   * Marks a span as error.
+   *
+   * @param span the span
+   * @param exception the exception that caused the error
+   */
+  public static void setSpanError(Span span, Throwable exception) {
+    span.setStatus(StatusCode.ERROR);
+    span.recordException(exception);
+  }
+
+  /**
+   * Marks a span as error.
+   *
+   * @param span the span
+   * @param description what went wrong
+   * @param exception the exception that caused the error
+   */
+  public static void setSpanError(Span span, String description, Throwable exception) {
+    span.setStatus(StatusCode.ERROR, description);
+    span.recordException(exception);
+  }
+
+  /**
+   * Injects the current context into a string map, which can then be added to HTTP headers or the
+   * metadata of an event.
+   */
+  public static Map<String, String> injectContext() {
+    Map<String, String> transport = new HashMap<>();
+    //noinspection ConstantConditions
+    GlobalOpenTelemetry.get()
+        .getPropagators()
+        .getTextMapPropagator()
+        .inject(
+            Context.current(),
+            transport,
+            (map, key, value) -> {
+              if (map != null) {
+                map.put(key, value);
+              }
+            });
+
+    return transport;
+  }
+
+  /**
+   * Extract the context from a string map, which you get from HTTP headers of the metadata of an
+   * event you're processing.
+   */
+  public static Context extractContext(Map<String, String> transport) {
+    Context current = Context.current();
+    //noinspection ConstantConditions
+    if (transport == null) {
+      return current;
+    }
+    // HTTP headers are case-insensitive. As we're using Map, which is case-sensitive, we need to
+    // normalize all the keys
+    Map<String, String> normalizedTransport =
+        transport.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    entry -> entry.getKey().toLowerCase(Locale.ROOT), Map.Entry::getValue));
+    return GlobalOpenTelemetry.get()
+        .getPropagators()
+        .getTextMapPropagator()
+        .extract(current, normalizedTransport, TEXT_MAP_GETTER);
+  }
+
+  /** Sets baggage items which are active in given block. */
+  @SuppressWarnings("NullAway")
+  public static <T> T setBaggage(Map<String, String> baggage, Callable<T> callable) {
```

Comment:
maybe `runWithBaggage`

Reply:
good point - I suggest `callWithBaggage` to make it consistent with the other methods.